model/project_ref.go

```diff
@@ -136,23 +136,24 @@ type ProjectRef struct {
 	ParsleyFilters    []parsley.Filter  `bson:"parsley_filters,omitempty" json:"parsley_filters,omitempty"`
 
 	// GitHubDynamicTokenPermissionGroups is a list of permission groups for GitHub dynamic access tokens.
-	GitHubDynamicTokenPermissionGroups GitHubDynamicTokenPermissionGroups `bson:"github_dynamic_token_permission_groups,omitempty" json:"github_dynamic_token_permission_groups,omitempty" yaml:"github_dynamic_token_permission_groups,omitempty"`
+	GitHubDynamicTokenPermissionGroups []GitHubDynamicTokenPermissionGroup `bson:"github_dynamic_token_permission_groups,omitempty" json:"github_dynamic_token_permission_groups,omitempty" yaml:"github_dynamic_token_permission_groups,omitempty"`
 
 	// GitHubPermissionGroupByRequester is a mapping of requester type to the user defined GitHub permission groups above.
 	GitHubPermissionGroupByRequester map[string]string `bson:"github_token_permission_by_requester,omitempty" json:"github_token_permission_by_requester,omitempty" yaml:"github_token_permission_by_requester,omitempty"`
 }
 
 // GitHubDynamicTokenPermissionGroup is a permission group for GitHub dynamic access tokens.
 type GitHubDynamicTokenPermissionGroup struct {
+	// Name is the name of the group.
 	Name string `bson:"name,omitempty" json:"name,omitempty" yaml:"name,omitempty"`
 	// Permissions are a key-value pair of GitHub token permissions to their permission level
 	Permissions github.InstallationPermissions `bson:"permissions,omitempty" json:"permissions,omitempty" yaml:"permissions,omitempty"`
```

Comment:
I think setting Permissions to `*github.InstallationPermissions` and then considering 
nil --> default/unset
empty --> no permissions 
might be easier to maintain. 

If you want to make it clearer and easier to use, you can add some helper functions like GitHubDynamicTokenPermissionGroup.IsDefualt() or isUnset() that checks if Permissions is nil, or even a GetGitHubDynamicTokenPermissionGroup() that returns a bool indicating if it's default or not.